torch/csrc/autograd/record_function.cpp

```diff
@@ -26,39 +29,85 @@ void popCallback() {
   }
   start_callbacks.pop_back();
   end_callbacks.pop_back();
-  has_callbacks = !start_callbacks.empty();
+  needs_inputs_flags.pop_back();
+  if (needs_inputs_flags.size() > 1) {
+    if (needs_inputs_flags.back()) {
+      record_function_state = RecordFunctionState::NAME_AND_INPUTS;
+    } else {
+      record_function_state = RecordFunctionState::NAME_ONLY;
+    }
+  } else {
```

Comment:
you technically don't need it as the list never goes empty

Reply:
But I'm checking not for emptiness, I'm checking if there're more than one (initial) element (that is, if we have callbacks), then we should check the last flag, otherwise it means that there're no callbacks left and we have to set NO_CALLBACKS.